tests/test_model/test_abstract_model.py

```diff
@@ -88,3 +98,36 @@ def test_pca(configspace_small, make_scenario):
     X_test, _ = get_X_y(configspace_small, n_samples, 10)
     with pytest.raises(ValueError, match="Feature mismatch.*"):
         model.predict_marginalized(X_test)
+
+def test_abstract_model_raises_without_istrained(configspace_small):
+    from dataclasses import dataclass
+    from ConfigSpace import ConfigurationSpace, Constant
+
+    @dataclass
+    class A:
+        a: int
+
+    def f() -> None:
+        return None
+
+    class DummyModel(AbstractModel):
+        def _train(self, X: np.ndarray, Y: np.ndarray):
+            self._is_trained = True
+            return self
+
+        @property
+        def is_trained(self) -> bool:
+            return getattr(self, "_is_trained", False)
+
+    @pytest.fixture
+    def configspace():
+        return ConfigurationSpace({
+            "cat": [True, False, None],
+            "othercat": [A(1), f],
+            "constant": Constant("constant", (24, 25)),
+        })
+    try:
+        model = DummyModel(configspace_small)
```

Comment:
is `configspace` used here /  what is `configspace_small`?

Reply:
it is merely a [fixture ](https://github.com/automl/SMAC3/blob/9d194754a5fed3ec48be06987cfc24ee99b76af5/tests/fixtures/configspace.py#L13) that i need to attempt instantiating the model. As long as it is a ConfigurationSpace object it doesn't really matter much what it contains.

Reply:
huups that snuck in there :)

Reply:
was from my if name main first demo that i copied over into the test :disappointed: